packages/mdl-snackbar/mdl-snackbar.scss

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import "mdl-animation/mixins";
+@import "mdl-theme/mixins";
+@import "mdl-typography/mixins";
+@import "./variables";
+
+/* postcss-bem-linter: define snackbar */
+.mdl-snackbar {
+  display: block;
+  display: flex;
+  position: fixed;
+  bottom: 0;
+  left: 50%;
+  justify-content: space-between;
+  transform: translate(0, 80px);
+  transition: transform .25s $mdl-animation-fast-out-linear-in-timing-function;
+  background-color: $snackbar-background-color;
+  cursor: default;
+  z-index: 3;
+  will-change: transform;
+  pointer-events: none;
+
+  @media (max-width: ($snackbar-tablet-breakpoint - 1)) {
+    left: 0;
+    width: 100%;
+    min-height: 48px;
+    max-height: 80px;
+  }
+
+  @media (min-width: $snackbar-tablet-breakpoint) {
+    min-width: 288px;
+    max-width: 568px;
+    transform: translate(-50%, 80px);
+    border-radius: 2px;
+  }
+
+  &--active {
+    transform: translate(0, 0);
+    pointer-events: auto;
+    transition: transform .25s $mdl-animation-linear-out-slow-in-timing-function;
+
+    @media (min-width: $snackbar-tablet-breakpoint) {
+      transform: translate(-50%, 0);
+    }
+  }
+  &__text {
```

Comment:
Since you're using flexbox, we can probably avoid using floats here in favor of some flex parameters.

Looking at these redlines:

![image](https://cloud.githubusercontent.com/assets/1185269/18395952/0d925c70-768f-11e6-99bb-e65b97f1482f.png)

We could do the following:

- Set left and right padding on `.mdl-snackbar` to `24px;`
- Set `align-items: center; justify-content: flex-start;` on `.mdl-snackbar`. This, combined with what's above, means you can remove the padding rules on both the text and action selectors completely.
- Set `margin-right: auto;` on `&__text`. This will push the element to the right of the text (the action) as far as possible without overflowing it (see [this article](https://medium.com/@samserif/flexbox-s-best-kept-secret-bd3d892826b6#.92b2t5m2z) for more information on the awesomeness that is flexbox's auto margins). Note that `margin-right` will have to be flipped to `margin-left` RTL layouts (e.g. `[dir="rtl"] &`). We do this for lists using a mixin - see https://github.com/google/material-design-lite/blob/master/packages/mdl-list/mdl-list.scss#L22. 
- Set `margin-left: 48px;` on `&__action`. Note that this will also probably have to be flipped.

Given these rules above, you can remove the `padding` and `float` rules from `&__text`, as well as the `float`, `text-align`, `padding`, and `align-self` rules from `&__action`. 

I'm sorry if some of this is nitpicky / a bit much but given that we're trying to create the canonical MD component library for v2, we have to be extremely careful that we adhere to the spec's guidelines as much as we can.